sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/E2ETelemetryItemValidation/LogsTests.cs

```diff
@@ -135,5 +141,98 @@ public void VerifyException(LogLevel logLevel, string expectedSeverityLevel)
+        [InlineData(false, false, true)]
+        [InlineData(false, true, true)]
+        public void ValidateLogSampling(bool logWithinActivityContext, bool enableSampling, bool sampledIn)
+        {
+            List<TelemetryItem>? telemetryLogItems = null;
+            List<TelemetryItem>? telemetryActivityItems = null;
+            MockPlatform platform = new MockPlatform();
+
+            platform.SetEnvironmentVariable(EnvironmentVariableConstants.ENABLE_LOG_SAMPLING, enableSampling ? "true" : "false");
+
+            // SETUP
+            var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                    .AddOpenTelemetry(options =>
+                    {
+                        options.AddAzureMonitorLogExporterForTest(platform, out telemetryLogItems);
+                    });
+            });
+
+            var uniqueTestId = Guid.NewGuid();
+
+            var logCategoryName = $"logCategoryName{uniqueTestId}";
+
+             var logger = loggerFactory.CreateLogger(logCategoryName);
+
+            var activitySourceName = $"activitySourceName{uniqueTestId}";
+            using var activitySource = new ActivitySource(activitySourceName);
+
+            var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource(activitySourceName)
+                .AddAzureMonitorTraceExporterForTest(out telemetryActivityItems)
+                .Build();
+
+            // ACT
+            if (logWithinActivityContext)
+            {
+                using var activity = activitySource.StartActivity(name: "SayHello");
+
+                if (!sampledIn && activity != null)
+                {
+                    activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;
+                }
+
+                logger.Log(
+                    LogLevel.Information,
+                    eventId: 0,
+                    exception: null,
+                    message: "Hello {name}.",
+                    args: new object[] { "World" });
+
+                activity?.Stop();
+            }
+            else
+            {
+                logger.Log(
+                   LogLevel.Information,
+                   eventId: 0,
+                   exception: null,
+                   message: "Hello {name}.",
+                   args: new object[] { "World" });
+            }
+
+            // flush logs
+            loggerFactory.Dispose();
+
+            if (enableSampling && logWithinActivityContext)
+            {
+                if (!sampledIn)
+                {
+                    Assert.NotNull(telemetryLogItems);
+                    Assert.Empty(telemetryLogItems);
+                }
+                else
+                {
+                    Assert.NotNull(telemetryLogItems);
+                    Assert.Single(telemetryLogItems);
+                }
+            }
+            else
+            {
+                Assert.NotNull(telemetryLogItems);
```

Comment:
Assert.NotNull is in all three if blocks. I think this should be moved up to line 217.